src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -881,13 +928,12 @@ public Builder withOkHttpClient(OkHttpClient okHttpClient) {
          * @return A new {@link Pinecone} instance configured based on the builder parameters.
          */
         public Pinecone build() {
-            PineconeConfig clientConfig = new PineconeConfig(apiKey);
-            clientConfig.setSourceTag(sourceTag);
-            clientConfig.validate();
+            PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig);
```

Comment:
We dont have to explicitly call .setSourceTag() if the sourceTag is part of the constructor

Reply:
Do users ever construct PineconeConfig directly? I see there are some constructor overloads